cfme/tests/automate/test_method.py

```diff
@@ -496,3 +496,70 @@ def test_embedded_method_selection():
         1718495
     """
     pass
+
+
+@pytest.mark.tier(1)
+def test_automate_state_method(klass):
+    """
+    You can pass methods as states compared to the old method of passing
+    instances which had to be located in different classes. You use the
+    METHOD:: prefix
+
+    Polarion:
+        assignee: ghubale
+        casecomponent: Automate
+        caseimportance: medium
+        initialEstimate: 1/4h
+        tags: automate
+        startsin: 5.6
+        testSteps:
+            1. Create an automate class that has one state.
+            2. Create a method in the class, make the method output
+               something recognizable in the logs
+            3. Create an instance inside the class, and as a Value for the
+               state use: METHOD::method_name where method_name is the name
+               of the method you created
+            4. Run a simulation, use Request / Call_Instance to call your
+               state machine instance
+        expectedResults:
+            1. Class created
+            2. Method created
+            3. Instance created
+            4. The method got called, detectable by grepping logs
+    """
+    state = fauxfactory.gen_alpha()
+
+    klass.schema.add_fields({'name': state, 'type': 'State'})
+
+    method = klass.methods.create(
```

Comment:
you probably should remove all the items you are creating using finalizers

Reply:
yes @jaryn I agree with you. 
- But we don't need cleanup for child items of ```domain, namespaces and klass``` because these are global fixtures and they have there own cleanup.
- Actually, **instance and methods are child items** of ```klass``` and if we delete klass then those items gets deleted.
Hence keeping as it is.